### DISCUSSION TITLE
Default field in schema object with type `object`

### DISCUSSION THREAD

**Discussion by mtovt** [OAI/OpenAPI-Specification#2948] [Q&A] [answered]
Hello! Is it possible to specify a default object property for object, using `default` field? 
For example I want to specify a default `foo` property in `Bar` object. I want to specify default `Foo` object only in the property of objects `Bar`. If it is possible, please provide an example.

```yml
openapi: 3.0.3
info:
  title: 'Example'
  version: 0.1.0
servers:
  - url: 'http://example.com'
paths:
  '/foo':
    delete:
      responses:
        '200':
          description: OK
components:
  schemas:
    Foo:
      type: object
      properties: 
        text:
          type: string
        number:
          type: number
    Bar:
      type: object
      properties:
        foo:
          allOf:
              - $ref: '#/components/schemas/Foo'
          default: "Is it possible to specify a default Foo object here?"
```

**Comment by flaksp** [comment, marked as answer]
Yes, it is possible.

OpenAPI 3.0 uses [JSON Schema (draft 5)](https://json-schema.org/specification-links.html#draft-5) spec to define schemas.

The `default` keyword is defined in [JSON Schema Validation spec, section 6.2](https://datatracker.ietf.org/doc/html/draft-wright-json-schema-validation-00#section-6.2):

> There are no restrictions placed on the value of this keyword.
>
> This keyword can be used to supply a default JSON value associated with a particular schema. It is RECOMMENDED that a default value be valid against the associated schema.

OpenAPI 3.0 clarifies semantics of the keyword in [section 4.7.25.1 of the spec](https://spec.openapis.org/oas/v3.0.0#properties):

> The default value represents what would be assumed by the consumer of the input as the value of the schema if one is not provided. Unlike JSON Schema, the value MUST conform to the defined type for the Schema Object defined at the same level. For example, if `type` is `string`, then `default` can be `"foo"` but cannot be `1`.

It means you may use `default` keyword to document what default behaviour of your API would be like if value is omited, for example.

Your example may be modified as:

```yaml
openapi: 3.0.3
info:
  title: Example
  version: 0.1.0
servers:
- url: http://example.com
paths:
  /foo:
    delete:
      responses:
        '200':
          description: OK
components:
  schemas:
    Foo:
      type: object
      properties:
        text:
          type: string
        number:
          type: number
    Bar:
      type: object
      properties:
        foo:
          allOf:
          - $ref: '#/components/schemas/Foo'
          default:
            text: abc
            number: 123
```

**Reply by mtovt** [reply]
Thank you for the explanation!

**Reply by geg-mopo3** [reply]
Doesn't it lead to generation of the class with the new name?
By me it would be BarFoo. However before the default application there were two classes: Bar and Foo.
And I haven't managed it to keep the previous name. title: 'Foo' didn't help.

**Comment by handrews** [comment]
This was answered and accepted, closing.
